File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DroneLauncher.java
package org.firstinspires.ftc.teamcode;

import com.qualcomm.robotcore.hardware.Gamepad;

public class DroneLauncher {
    private Robot robot = null;
    private Gamepad gamepad = null;
    private double pos_hold = 1;
    private double pos_release = 0.8;
    public DroneLauncher(Robot robot, Gamepad gamepad)
    {
            this.robot = robot;
            this.gamepad = gamepad;
    }

    public void launchDrone()
    {
        if (gamepad.b) {
            robot.servoDrone.setPosition(pos_release);
        }
        holdDrone();
    }

    public void holdDrone()
    {
        if (gamepad.a) {
            robot.servoDrone.setPosition(pos_hold);
        }
    }

}

File: TeamCode/src/main/kotlin/org/firstinspires/ftc/teamcode/DroneLauncher.kt
package org.firstinspires.ftc.teamcode

import com.qualcomm.robotcore.eventloop.opmode.OpMode
import com.qualcomm.robotcore.hardware.HardwareDevice

class DroneLauncher(private var op: OpMode, private var motor: HardwareDevice) {
    @Suppress("MemberVisibilityCanBePrivate")
    var isArmed: Boolean = false
        private set;

    /**
     * @return "true" if it was successfully armed or "false" if it was already armed
     */
    fun arm(): Boolean {
        TODO("Not yet implemented")
//        return if (!isArmed) {
//            if (false /** catastrophic error */) throw Exception("this will never be called, catastrophic error");
//            true;
//        } else {
//            false;
//        }
    }

//    /**
//     * @return "true" if it was successfully armed or "false" if it was already armed
//     */
//    public fun armAsync(): Boolean { }

    /**
     * @return "true" if it was successfully disarmed or "false" if it was already disarmed
     */
    fun disarm(): Boolean {
        TODO("Not yet implemented")
        return if (isArmed) {
            if (false /** catastrophic error */) throw Exception("this will never be called, catastrophic error");
            true;
        } else {
            false;
        }
    }

//    /**
//     * @return "true" if it was successfully disarmed or "false" if it was already disarmed
//     */
//    public fun disarmAsync(): Boolean { }

    fun launch(launchWithoutArming: Boolean?): Boolean {
        TODO("Not yet implemented")
        return if (isArmed || launchWithoutArming == true) {
            if (false /** catastrophic error */) throw Exception("this will never be called, catastrophic error");
            true;
        } else {
            false;
        }
    }

//    public fun launchAsync(launchWithoutArming: Boolean?): Boolean { }
}